### PR TITLE
Implement AVS list error handling

### PR DESCRIPTION
**Pull request by Zishan-7**
- Added default sorting indication
- Added error handling
- Modified table empty state

This is how the errors will look:

Screenshot 2024-07-23 at 8 26 33 PM

When there is no AVS found for a search query the table headers will not be shown

Screenshot 2024-07-23 at 8 28 42 PM

### CODE REVIEW

File: src/avs/AVSList.jsx [hunk 1]
```diff
@@ -171,122 +172,131 @@ export default function AVSList() {
...
+
+      {!state.error && (
+        <div className="flex flex-1 flex-col rounded-lg border border-outline bg-content1 text-sm">
+          <Table
+            aria-label="AVS list"
+            layout="fixed"
+            removeWrapper
+            classNames={{
+              base: 'overflow-x-auto h-full',
+              table: 'h-full'
```
Comment (rubo) [inline src/avs/AVSList.jsx:202]: Not related to this PR, but I noticed a weird thing and couldn't find an "official" way to get rid of it. Changed the colors, so you can spot it:

image

This empty space is on each table. @vincenthongzy, you need to apply this to your tables as well.

```suggestion
              table: 'h-full',
              thead: '[&>tr:last-child]:hidden'
```

**Review by rubo — CHANGES_REQUESTED** [review]
It shows the second screen with "No result found..." instead of an error. We need an error message regardless of whether it's a search or not.

The default sorting shows up fine.

File: src/avs/AVSList.jsx (hunk 1 above)
Comment (Zishan-7) [reply src/avs/AVSList.jsx]: Added suggested change to the table component.

File: src/avs/AVSList.jsx [hunk 2]
```diff
@@ -90,10 +91,10 @@ export default function AVSList() {
...
 
         abortController.current = null;
-      } catch (error) {
-        if (error.name !== 'AbortError') {
+      } catch (e) {
+        if (e.name !== 'AbortError') {
           dispatch({
-            error: 'Failed to fetch AVS data',
+            error: handleServiceError(e),
             isFetchingAvsData: false
```
Comment (rubo) [inline src/avs/AVSList.jsx:98]: Minor. Please follow the naming convention -- `isFetchingAVSData.` As this view loads only AVS data, we can keep it short `isFetching` or `isFetchingData`...

**Review by rubo — CHANGES_REQUESTED** [review]
We made a mistake here. Please see https://github.com/NethermindEth/restaking-dashboard-backend/pull/95.

So, if there's an error, we display an error message using the `ErrorMessage` component as you have on the first screen. But, if AVS is not found, it returns an empty collection, and you show the second screen.

**Review by rubo — APPROVED** [review]

**Pull request merged**